torchmetrics/metric.py

```diff
@@ -258,27 +269,55 @@ def sync(
                 only when running in a distributed setting.
             distributed_available: Function to determine if we are running inside a distributed setting
 
-        Returns:
-            cache: A dictionary containing the local metric states. The cache will be empty if sync didn't happen.
         """
+        if self.is_synced and should_sync:
+            raise TorchMetricsUserError("The Metric has already been synced.")
+
         is_distributed = distributed_available() if callable(distributed_available) else None
+
         if not should_sync or not is_distributed:
-            return {}
+            return
+
         if dist_sync_fn is None:
             dist_sync_fn = gather_all_tensors
+
         # cache prior to syncing
-        cache = {attr: getattr(self, attr) for attr in self._defaults}
+        self._cache = {attr: getattr(self, attr) for attr in self._defaults}
+
         # sync
         self._sync_dist(dist_sync_fn, process_group=process_group)
-        return cache
+        self.is_synced = True
+
+    def unsync(self, should_unsync: bool = True) -> None:
+        """
+        Unsync function for manually controlling when metrics states should be reverted back to their local states.
+
+        Args:
+            should_unsync: Whether to perform unsync
+
+        """
+        if not should_unsync:
+            return
+
+        if self.is_synced:
+            if self._cache is None:
```

Comment:
to confirm, after users call `sync`, the only valid actions they can take afterward are either `reset` or `unsync`? 
since this is now a state machine, is this enforced in all spots?

Reply:
Yes, and with the new API, compute will be authorised.